CAP/gap/OppositeCategory.gi

```diff
@@ -89,6 +91,49 @@ BindGlobal( "CAP_INTERNAL_INSTALL_OPPOSITE_ADDS_FROM_CATEGORY",
     
     recnames := AsSortedList( RecNames( CAP_INTERNAL_METHOD_NAME_RECORD ) );
     
+    ## No support for twocells
+    recnames := Difference( recnames,
+                        [ "HorizontalPreCompose",
+                          "HorizontalPostCompose",
+                          "VerticalPreCompose",
+                          "VerticalPostCompose",
+                          "IdenticalTwoCell" ] );
+    
+    if only_primitive_operations then
+        list_of_installed_operations := ListPrimitivelyInstalledOperationsOfCategory( category );
+    else
+        list_of_installed_operations := ListInstalledOperationsOfCategory( category );
+    fi;
+    
+    operations_of_homomorphism_structure :=
+      [ "DistinguishedObjectOfHomomorphismStructure",
+        "HomomorphismStructureOnObjects",
+        "HomomorphismStructureOnMorphisms",
+        "HomomorphismStructureOnMorphismsWithGivenObjects",
+        "InterpretMorphismAsMorphismFromDistinguishedObjectToHomomorphismStructure",
+        "InterpretMorphismAsMorphismFromDistinguishedObjectToHomomorphismStructureWithGivenObjects",
+        "InterpretMorphismFromDistinguishedObjectToHomomorphismStructureAsMorphism",
+        ];
+    
+    operations_of_external_hom_in_matrix_category :=
+      [ "BasisOfExternalHom",
+        "CoefficientsOfMorphism",
+        ];
+    
+    operations_of_external_hom_in_fin_sets :=
+      [ "MorphismsOfExternalHom", ## still in Toposes but should be moved to CAP
+        ];
+    
+    if HasRangeCategoryOfHomomorphismStructure( category ) and
+       ( not IsEmpty( Intersection( list_of_installed_operations, operations_of_homomorphism_structure ) ) or
+         not IsEmpty( Intersection( list_of_installed_operations, operations_of_external_hom_in_matrix_category ) ) or
+         not IsEmpty( Intersection( list_of_installed_operations, operations_of_external_hom_in_fin_sets ) ) ) then
+        
+        SetRangeCategoryOfHomomorphismStructure( opposite_category, RangeCategoryOfHomomorphismStructure( category ) );
+        SetIsEquippedWithHomomorphismStructure( opposite_category, true );
```

Comment:
I don't think this fits into the new system saying "the range category is type information which should be explicitly provided because we don't want to guess what the user wants". For example, "BasisOfExternalHom" and "CoefficientsOfMorphism" might be installed in the underlying category but not be used for deriving the homomorphism structure, but this code would still blindly use its range (and trigger or not trigger some derivation based on this guess).

I suggest the following: keep the case of `operations_of_homomorphism_structure` (if the user wants us to dualize the hom structure, that's fine), but drop the other special cases. If one wants to have a hom structure in those special cases, one must set the intended range explicitly. If this happens at many places, we could add a convenience function which accepts the range category as an additional argument.

Reply:
With my suggestion the tests in CategoricalTowers (excluding Algebroids and FunctorCategories which I have not tested) seem to still succeed.

Reply:
Done.

Reply:
FunctorCategories now fails miserably :) I hope I can get it fixed.

Reply:
False alarm: After setting FinSetsForCAP to the adjusted branch also FunctorCategories passed 👍

Reply:
Reminder: The above testing refers to PR #1240 (the "next" PR), not to this one.